[MRG] fixes issue #6346: changed default percentile param

vincentpham1991 opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#Contributing-Pull-Requests
-->
#### Reference Issue
<!-- Example: Fixes #1234 -->
#6346

#### What does this implement/fix? Explain your changes.
Changed default param for percentile to a more reasonable default value.

#### Any other comments?


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

Commit:
default param change

nelson-liu reviewed (commented):
This probably needs a whatsnew entry?

jnothman commented:
As suggested in #6346, it's not this simple, and requires some deprecation process. Imagine, for instance, a user had relied on this default when selecting a different `mode`.

TomDLT commented:
> it requires some deprecation process

or at the very least, as stated in #6346, we would need to document that as a breaking change.

lesteve commented:
> As suggested in #6346, it's not this simple, and requires some deprecation process

@vincentpham1991 have a look at http://scikit-learn.org/stable/developers/contributing.html#deprecation if you want to implement this.

Commit:
added deprecation warning and what's new

vincentpham1991 commented:
Thank you everyone for the comments and help and sorry for the delay. I addressed the comments above, please let me know what else needs to be changed.

Commit:
merge conflict

Commit:
original whats_new

Commit:
add what's new

jnothman commented:
You've introduced some mess in your commit history; and for deprecation, the default must remain the same for now, and the warning should be silenced if the user enters a non-default.

thomasjpfan commented:
As noted in https://github.com/scikit-learn/scikit-learn/issues/6346#issuecomment-991984988, I do not think that changing the default to 10 will resolve the issue. With that in mind, I am closing this PR. Thank you for working on this PR!
